### PR TITLE
support authenticating with a service principal in Linux

### PR DESCRIPTION
**Pull request by afeld**
It appears issues got turned off in this repository, but I had described the background in [#13864](https://github.com/MicrosoftDocs/azure-devops-docs/issues/13864#issuecomment-1928675057): I was trying to set up a Pipeline agent in a Docker container on Linux using a service principal, and it wasn't clear how to authenticate.

This pull request updates the instructions and code for Linux to support authentication with a service principal, which I was able to do locally. I don't know Powershell so I didn't touch that part of the page, but it should be possible there too.

Thanks!

### CODE REVIEW

**Comment by afeld** [comment]
@microsoft-github-policy-service agree

File: docs/pipelines/agents/docker.md [hunk 1]
```diff
@@ -421,11 +437,25 @@ You can control the agent name, the agent pool, and the agent work directory by
...
+### Authentication
+
+One of the following is required:
+
+- If using a PAT: `AZP_TOKEN`
+- If using a service principal: `AZP_CLIENTID`, `AZP_CLIENTSECRET`, and `AZP_TENANTID`
+
+### Authorization
+
+The token or service principal must have the **Agent Pools (read, manage)** scope at the Organization level of `AZP_URL`. If using a PAT, the token must be created by a user who has permission to [configure agents](pools-queues.md#create-agent-pools).
```
Comment (afeld) [inline docs/pipelines/agents/docker.md:457]: Not 100% confident I got this part right.

**Comment by Jak-MS** [comment]
@steved0x 
Can you review the proposed changes? 

**Important**: When the changes are ready for publication, adding a `#sign-off` comment is the best way to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Review by steved0x — APPROVED** [review]

**Comment by steved0x** [comment]
#sign-off

**Comment by afeld** [comment]
I’m not seeing the change on [the live page](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/docker?view=azure-devops) yet - is that expected?

**Comment by Jak-MS** [comment]
hi @afeld - this PR just missed the last sync/publish and will go live with the next one.  @steved0x  - for visibility.

**Pull request merged**